openmdao/error_checking/check_config.py

```diff
@@ -506,6 +507,60 @@ def _check_missing_recorders(problem, logger):
     logger.warning(msg)
 
 
+def _check_unserializable_options(problem, logger, check_recordable=True):
+    """
+    Check if there are any options that are not serializable, and therefore won't be recorded.
+
+    Parameters
+    ----------
+    problem : <Problem>
+        The problem being checked.
+    logger : object
+        The object that manages logging output.
+    check_recordable : bool
+        If False, warn about all unserializable options.
+        If True, warn only about unserializable options that do not have 'recordable' set to False.
+    """
+    from openmdao.recorders.case_recorder import PICKLE_VER
+
+    def _check_opts(obj):
+        if obj:
+            for key, val in obj.options.items():
+                try:
+                    pickle.dumps(val, PICKLE_VER)
+                except Exception:
+                    if obj.options._dict[key]['recordable']:
+                        msg = f"{obj.msginfo}: option '{key}' is not serializable " \
+                              "(cannot be pickled) but 'recordable=False' has not been set. " \
+                              f"No options will be recorded for this {obj.__class__.__name__} " \
+                              "unless 'recordable' is set to False for this option."
+                        logger.warning(msg)
+                    elif not check_recordable:
+                        msg = f"{obj.msginfo}: option '{key}' is not serializable " \
+                              "(cannot be pickled) and will not be recorded."
+                        logger.warning(msg)
+
+    # check options for all for Systems and Solvers
+    for system in problem.model.system_iter(include_self=True, recurse=True):
+        _check_opts(system)
+        _check_opts(system.nonlinear_solver)
+        _check_opts(system.linear_solver)
```

Comment:
A couple more to check:

`nonlinear_solver.linear_solver`
`nonlinear_solver.linesearch`